saleor/graphql/order/filters.py

```diff
@@ -79,12 +81,30 @@ def filter_status(qs, _, value):
 
 
 def filter_customer(qs, _, value):
-    qs = qs.filter(
-        Q(user_email__ilike=value)
-        | Q(user__email__trigram_similar=value)
-        | Q(user__first_name__trigram_similar=value)
-        | Q(user__last_name__trigram_similar=value)
-    )
+    try:
+        validate_email(value)
+        qs = qs.filter(Q(user_email__iexact=value) | Q(user__email__iexact=value))
+    except ValidationError:
+        try:
+            first, last = value.split(" ", 1)
+            qs = qs.alias(
+                user_full_name=FlatConcat(
+                    "user__first_name",
+                    Value(" "),
+                    "user__last_name",
+                )
+            ).filter(
+                Q(user_full_name__iexact=value)
+                | Q(user_full_name__iexact=f"{last} {first}")
+            )
+        except ValueError:
+            qs = qs.filter(
+                Q(user_email__ilike=value)
+                | Q(user__email__ilike=value)
+                | Q(user__first_name__ilike=value)
+                | Q(user__last_name__ilike=value)
+            )
```

Comment:
I do not like these nested `try except` parts. In `try` you should have only the part that might raise an error. If you want to call something when error is not raised you should use `else` part.
Maybe we should split it into multiple methods like that:

```suggestion
def filter_customer(qs, _, value):
    try:
        validate_email(value)
    except ValidationError:
       return  _filter_customer_by_full_name(value)
   # filter by user email if the provided value is an email
    return  qs.filter(Q(user_email__iexact=value) | Q(user__email__iexact=value))


def _filter_customer_by_full_name(value):
        try:
            first, last = value.split(" ", 1)
        except ValueError:
            return qs.filter(
                Q(user_email__ilike=value)
                | Q(user__email__ilike=value)
                | Q(user__first_name__ilike=value)
                | Q(user__last_name__ilike=value)
            )
        return  qs.alias(
                user_full_name=FlatConcat(
                    "user__first_name",
                    Value(" "),
                    "user__last_name",
                )
            ).filter(
                Q(user_full_name__iexact=value)
                | Q(user_full_name__iexact=f"{last} {first}")
            )
```
WDYT?